### ISSUE TITLE
[Worker support] Cannot start the worker if the HTML page is not in the same domain as pyscript.js

### ISSUE THREAD

**Issue by antocuni** [pyscript/pyscript#1476]
This is a serious problem: at the moment of writing, we can use `execution_mode = "workers"` ONLY if the web page is served by the same domain as pyscript.js (or to be more precise, interpretr_worker.js, which is its sibling).
In practice, it means that you cannot use `execution_mode = "workers"` if you are getting pyscript.js from a CDN, which is the common case.
We never noticed it so far because in tests this is always the case (as everything is served by localhost).

More context in this discord chat:
https://discord.com/channels/972017612454232116/1108388383694258260

TL;DR: when we call `new Worker("https://.../interpreter_worker.js")`, the js file must be served by the same domain as the main page. @hoodmane can probably write a better summary because I'm not 100% sure to have understood correctly.

**Comment by bugzpodder** [comment]
Not claiming that this is the preferred way for this problem, but you can do something like:

```
const blob = new Blob([workerSrc], { type: "application/javascript" });
                  const url = window.URL || window.webkitURL;
                  const blobUrl = url.createObjectURL(blob);
                  new Worker(blobUrl);
                  ```

**Comment by bugzpodder** [comment]
Also, on a related note I was seeing an error locally for 
```
When execution_thread is "worker", the site must be cross origin isolated
```
when the check crossOriginIsolated = false, so I was blocked from doing any testing locally.

I am just serving the worker and py-script from localhost, and just removing this check seem to work fine for me.

**Comment by WebReflection** [comment]
**More findings**

Because we're basically playing indirectly with code evaluation, we should be aware (and make our users aware) that using the blob technique might fail in case [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src) rules don't allow evaluation.

**Comment by bugzpodder** [comment]
partytown might be of interest to some: https://partytown.builder.io/trade-offs

**Comment by WebReflection** [comment]
Should  close this as won't fix?

**Comment by WebReflection** [comment]
we currently fetch and run worker code as a Blob these days so I think this can be closed.
